plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -128,10 +140,38 @@ export class HogTransformerService {
                 const transformationsFailed: string[] = event.properties?.$transformations_failed || []
                 const transformationsSkipped: string[] = event.properties?.$transformations_skipped || []
 
+                // Get states for all functions to check if any are disabled - only if feature flag is enabled
+                let states: Record<string, { state: HogWatcherState }> = {}
+                if (this.hub.TRANSFORM_EVENT_HOG_WATCHER_ENABLED) {
+                    const timer = transformEventHogWatcherLatency.startTimer()
+                    states = await this.hogWatcher.getStates(teamHogFunctions.map((hf) => hf.id))
+                    const durationSeconds = timer()
+                    transformEventHogWatcherLatency.observe({ operation: 'getStates' }, durationSeconds * 1000) // Convert seconds to milliseconds
+                }
+
                 // For now, execute each transformation function in sequence
                 for (const hogFunction of teamHogFunctions) {
                     const transformationIdentifier = `${hogFunction.name} (${hogFunction.id})`
 
+                    // Check if the function is disabled via hogWatcher - skip if feature flag is disabled
+                    if (this.hub.TRANSFORM_EVENT_HOG_WATCHER_ENABLED) {
```

Comment:
I'm wondering if we could ditch this part for the first part of testing. That way if we have any bugs issues then we will know about it and not just disable a bunch of transforms that shouldn't be disabled. 

We could still log a metric "would have dropped" or something